src/Raven.Server/Documents/Expiration/ExpirationStorage.cs

```diff
@@ -176,22 +169,18 @@ public record ExpiredDocumentsOptions
 
                                     if (allExpired)
                                     {
-                                        expiredDocs.Add((clonedId, id));
+                                        expired.Enqueue(new ExpiredDocumentInfo(ticksAsSlice, clonedId, id));
                                         totalCount++;
                                     }
                                 }
                             } while (multiIt.MoveNext() 
-                                     && expiredDocs.Count + count < options.AmountToTake 
+                                     && expired.Count < options.AmountToTake 
```

Comment:
I see that `expired` is explicitly cleared on tx dispose, only if the tx was committed. Does it mean that if we fail to commit for some reason we might loop here?

Reply:
Okay so without this fix we could have some inconsistency issues (that we could skip something to expire or something like that)?

Reply:
With the new implementation, yes. (by using a `queue`).

Reply:
No, I was referring to the old (current) implementation

Reply:
In the current implementation we don't have any inconsistency because we always process everything that we got.